# 8. Модуль NumPy. Операции с векторами

### Общий код для выполнения заданий 8.4 - 8.6

In [1]:
import numpy as np
a = np.array([23, 34, 27])
b = np.array([-54, 1,  46])
c = np.array([46, 68, 54])

### Задание 8.4

Векторы в геометрии, которые параллельны друг другу, называются сонаправленными. На практике есть несколько способов определить сонаправленность векторов. Один из них: сумма длин сонаправленных векторов должна быть равной длине суммы двух векторов.

С помощью этого критерия найдите пару сонаправленных векторов из векторов a, b и c.

In [2]:
def is_collinear(vec_1, vec_2):
    return (np.linalg.norm(vec_1) + np.linalg.norm(vec_2)) == np.linalg.norm(vec_1 + vec_2)

In [3]:
is_collinear(a, b)

False

In [4]:
is_collinear(a, c)

True

In [5]:
is_collinear(b, c)

False

Ответ: a и c

### Задание 8.5

Найдите пару векторов, расстояние между которыми больше 100.

In [8]:
print(f'Растояние между векторами a и b: {np.linalg.norm(a-b)}')
print(f'Растояние между векторами a и c {np.linalg.norm(a-c)}')
print(f'Растояние между векторами b и c: {np.linalg.norm(b-c)}')

Растояние между векторами a и b: 85.901105930017
Растояние между векторами a и c 49.13247398615299
Растояние между векторами b и c: 120.6358155772986


Ответ: b и c

### Задание 8.6

Найдите пару перпендикулярных векторов с помощью скалярного произведения (оно должно быть равно нулю).

In [9]:
def is_ortogonal(vec_1, vec_2):
    return np.dot(vec_1, vec_2) == 0

print(f'Векторы a и b перпендикулярны: {is_ortogonal(a, b)}')
print(f'Векторы a и c перпендикулярны: {is_ortogonal(a, c)}')
print(f'Векторы b и c перпендикулярны: {is_ortogonal(b, c)}')

Векторы a и b перпендикулярны: False
Векторы a и c перпендикулярны: False
Векторы b и c перпендикулярны: False


Ответ: такой пары нет

# Общий код для заданий 8.7 - 8.10

In [17]:
from mystery import mystery

### Задание 8.7
Найдите минимальное значение в массиве mystery

In [18]:
mystery.min()

5

Ответ: 5

### Задание 8.8

Найдите среднее значение массива. Ответ не округляйте.

In [19]:
mystery.mean()

490.368

Ответ: 490.368

### Задание 8.9

Найдите медиану массива.

In [22]:
np.median(mystery)

491.0

Ответ: 491

### Задание 8.10

Найдите стандартное отклонение значений в массиве. Ответ округлите до сотых.

In [24]:
np.round(np.std(mystery), 2)

289.01

Ответ: 289.01